### ISSUE TITLE
functional case coverage does not get the right coverage

### ISSUE THREAD

**Issue by JunelanMe** [Codeception/Codeception#3609]
#### What are you trying to achieve?

the coverage should be  (code line trigger by case )/(all php line code)
#### What do you get instead?

(code line trigger by case )/(all action ,module ,etc function line )
###### #Output for codeception 2.0.4
- Codeception version: 2.0.15 
- PHP Version: php 5.4.45 
- Operating System: linux
- Installation type: Composer
- List of installed packages (`composer show`)
- Suite configuration:
  ####codeception.yml config####

``` yml
coverage:
    enabled: true
    remote: true
    c3_url: 'https://www.vagrant.foxitarchive.com/index-test.php'
    include:
             - ../../../foxitarchiveapp/assets/*.php
             - ../../../foxitarchiveapp/common/*.php
             - ../../../foxitarchiveapp/components/*.php
             - ../../../foxitarchiveapp/config/*.php
             - ../../../foxitarchiveapp/controllers/*.php
             - ../../../foxitarchiveapp/models/*.php
    exclude:
             - ../../../foxitarchiveapp/controllers/DevController.php
             - ../../../foxitarchiveapp/controllers/SettingController.php
```

![urly4 q5z qc7ma qrm5](https://cloud.githubusercontent.com/assets/11918045/19385045/34c16de2-923f-11e6-9a4e-86b8482facbb.png)

as the pic show , the coverage in codeption 2.0 is case trigger line / all php code line
###### Output for codeception 2.2.4

```
* Codeception version:  PHP_CodeCoverage 2.2.4
* PHP Version: php PHP 7.0.11,
* Operating System: linux
* Installation type: Composer
* List of installed packages (`composer show`)
* Suite configuration:
```
#### codeception.yml config#### same as the 2.0.4

``` yml
coverage:
    enabled: true
    remote: true
    c3_url: 'https://www.vagrant.foxitarchive.com/index-test.php'
    include:
             - ../../../foxitarchiveapp/assets/*.php
             - ../../../foxitarchiveapp/common/*.php
             - ../../../foxitarchiveapp/components/*.php
             - ../../../foxitarchiveapp/config/*.php
             - ../../../foxitarchiveapp/controllers/*.php
             - ../../../foxitarchiveapp/models/*.php
    exclude:
             - ../../../foxitarchiveapp/controllers/DevController.php
             - ../../../foxitarchiveapp/controllers/SettingController.php
```

```
[root@c71 webapp]# php composer.phar show
Cannot load Xdebug - it was already loaded
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
behat/gherkin                     v4.4.4  Gherkin DSL parser for PHP 5.3
bower-asset/bootstrap             v3.3.4  The most popular front-end framework for developing responsive, mobile first projects on the web.
bower-asset/bootstrap-tagsinput   0.4.2   jQuery plugin providing a Twitter Bootstrap user interface for managing tags.
bower-asset/fontawesome           v4.5.0  Font Awesome
bower-asset/jquery                2.1.3
bower-asset/jquery-ui             1.11.4
bower-asset/jquery.inputmask      3.2.7   jquery.inputmask is a jquery plugin which create an input mask.
bower-asset/punycode              v1.3.2
bower-asset/typeahead.js          v0.10.5
bower-asset/underscore            1.8.3
bower-asset/yii2-pjax             v2.0.4
cebe/markdown                     1.0.2   A super fast, highly extensible markdown parser for PHP
codeception/c3                    2.0.8   CodeCoverage collector for Codeception
codeception/codeception           2.2.4   BDD-style testing framework
codeception/specify               0.4.3   BDD code blocks for PHPUnit and Codeception
codeception/verify                0.3.2   BDD assertion library for PHPUnit
doctrine/annotations              v1.2.3  Docblock Annotations Parser
doctrine/instantiator             1.0.4   A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                    v1.0.1  Base library for a lexer that can be used in Top-Down, Recursive Descent Parsers.
endroid/qrcode                    1.5.0   Endroid QR Code
ezyang/htmlpurifier               v4.6.0  Standards compliant HTML filter written in PHP
facebook/webdriver                1.1.3   A PHP client for WebDriver
fxp/composer-asset-plugin         v1.1.3  NPM/Bower Dependency Manager for Composer
fzaninotto/faker                  v1.4.0  Faker is a PHP library that generates fake data for you.
graylog2/gelf-php                 1.2.0   A php implementation to send log-messages to a GELF compatible backend like Graylog2.
guzzlehttp/guzzle                 6.2.1   Guzzle is a PHP HTTP client library
guzzlehttp/promises               1.2.0   Guzzle promises library
guzzlehttp/psr7                   1.3.1   PSR-7 message implementation
guzzlehttp/ringphp                1.0.6   Provides a simple API and specification that abstracts away the details of HTTP into a single PHP function.
guzzlehttp/streams                3.0.0   Provides a simple abstraction over streams of data
hybridauth/hybridauth             v2.4.1  Open source social sign on PHP library.
illuminate/support                v4.2.17
intervention/image                1.6.5   Image handling and manipulation library with support for Laravel 4 integration
jasig/phpcas                      1.3.3   Provides a simple API for authenticating users against a CAS server
mongodb/mongodb                   1.0.2   MongoDB driver library
mrclay/minify                     2.3.0   Minify is a PHP5 app that helps you follow several rules for client-side performance. It combines multiple CSS or Javascript files, removes unnecess...
myclabs/deep-copy                 1.2.1   Create deep copies (clones) of your objects
natxet/CssMin                     v3.0.4  Minifying CSS
paypal/adaptivepayments-sdk-php   v3.9.0  PayPal adaptivepayments SDK for PHP
paypal/sdk-core-php               v3.2.0  PayPal Core SDK for PHP
pdepend/pdepend                   2.0.6   Official version of pdepend to be handled with Composer
phpdocumentor/reflection-docblock 2.0.4
phpmd/phpmd                       2.2.1   PHPMD is a spin-off project of PHP Depend and aims to be a PHP equivalent of the well known Java tool PMD.
phpspec/php-diff                  v1.0.2  A comprehensive library for generating differences between two hashable objects (strings or arrays).
phpspec/prophecy                  v1.3.1  Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage         2.2.4   Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator         1.4.1   FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template         1.2.1   Simple template engine.
phpunit/php-timer                 1.0.8   Utility class for timing
phpunit/php-token-stream          1.4.8   Wrapper around PHP's tokenizer extension.
phpunit/phpunit                   4.8.27  The PHP Unit Testing framework.
phpunit/phpunit-mock-objects      2.3.8   Mock Object library for PHPUnit
psr/http-message                  1.0.1   Common interface for HTTP messages
psr/log                           1.0.0   Common interface for logging libraries
react/promise                     v2.2.0  A lightweight implementation of CommonJS Promises/A for PHP
rmrevin/yii2-fontawesome          2.13.0  Asset Bundle for Yii2 with Font Awesome
sebastian/comparator              1.1.1   Provides the functionality to compare PHP values for equality
sebastian/diff                    1.4.1   Diff implementation
sebastian/environment             1.3.8   Provides functionality to handle HHVM/PHP environments
sebastian/exporter                1.2.0   Provides the functionality to export PHP variables for visualization
sebastian/global-state            1.0.0   Snapshotting of global state
sebastian/recursion-context       1.0.0   Provides functionality to recursively process PHP variables
sebastian/version                 1.0.4   Library that helps with managing the version number of Git-hosted PHP projects
skeeks/yii2-assets-auto-compress  1.2.1   Automatically compile and merge files js + css
squizlabs/php_codesniffer         1.5.6   PHP_CodeSniffer tokenises PHP, JavaScript and CSS files and detects violations of a defined set of coding standards.
stripe/stripe-php                 v1.18.0 Stripe PHP Library
swiftmailer/swiftmailer           v5.4.0  Swiftmailer, free feature-rich PHP mailer
symfony/browser-kit               v3.1.4  Symfony BrowserKit Component
symfony/config                    v3.1.4  Symfony Config Component
symfony/console                   v3.1.4  Symfony Console Component
symfony/css-selector              v3.1.4  Symfony CssSelector Component
symfony/dependency-injection      v3.1.4  Symfony DependencyInjection Component
symfony/dom-crawler               v3.1.4  Symfony DomCrawler Component
symfony/event-dispatcher          v3.1.4  Symfony EventDispatcher Component
symfony/filesystem                v3.1.4  Symfony Filesystem Component
symfony/finder                    v3.1.4  Symfony Finder Component
symfony/polyfill-mbstring         v1.2.0  Symfony polyfill for the Mbstring extension
symfony/yaml                      v3.1.4  Symfony Yaml Component
tedivm/jshrink                    v1.1.0  Javascript Minifier built in PHP
yiisoft/yii2                      2.0.9   Yii PHP Framework Version 2
yiisoft/yii2-bootstrap            2.0.3   The Twitter Bootstrap extension for the Yii framework
yiisoft/yii2-codeception          2.0.3   The Codeception integration for the Yii framework
yiisoft/yii2-composer             2.0.4   The composer plugin for Yii extension installer
yiisoft/yii2-debug                2.0.3   The debugger extension for the Yii framework
yiisoft/yii2-faker                2.0.3   Fixture generator. The Faker integration for the Yii framework.
yiisoft/yii2-gii                  2.0.3   The Gii extension for the Yii framework
yiisoft/yii2-jui                  2.0.3   The Jquery UI extension for the Yii framework
yiisoft/yii2-mongodb              2.1.1   MongoDb extension for the Yii framework
yiisoft/yii2-swiftmailer          2.0.3   The SwiftMailer integration for the Yii framework
zircote/swagger-php               0.9.6   Swagger-PHP library implementing the swagger.wordnik.com specification to describe web services, operations/actions and models enabling a uniform me...
```

As the screenshot show below ,the coverage in latest version is (code line trigger by case )/(all action ,module ,etc function line ) , it is most higher than 2.0.15 . Some lines which case does not trigger   developer's code is not counted as the denominator . It seems not reasonable

[root@c71 webapp]#
![2 2 4](https://cloud.githubusercontent.com/assets/11918045/19424037/889b4206-9458-11e6-8841-789432263307.png)

I select one action of a controller  as follow , the action has 20 lines . But the coverage show 0 lines . 
![archiveindex](https://cloud.githubusercontent.com/assets/11918045/19424254/22904f4a-945a-11e6-91f5-569faecd04f0.png)

![action](https://cloud.githubusercontent.com/assets/11918045/19424027/762e11b6-9458-11e6-920c-3ac2ecd3c8c0.png)

**Comment by Naktibalda** [comment]
All your screenshots were made with PHP_CodeCoverage 2.0.15, but your `composer show` shows that you are using PHP_CodeCoverage 2.2.4.

Anyway, most likely it is an issue of xdebug.

**Comment by DavertMik** [comment]
yep, I think in latest CodeCoverage they changed the metrics so to have better detailed (lower) coverage.
Sorry I'm not expert in code coverage

**Comment by JunelanMe** [comment]
@Naktibalda 
Sorry ,I updated the wrong pic bellow composer show , I have already updated them   . The two pics below composer show  should be following . 
![action](https://cloud.githubusercontent.com/assets/11918045/19424088/ff363fd8-9458-11e6-882f-6f46de368678.png)

![2 2 4](https://cloud.githubusercontent.com/assets/11918045/19424089/04398af8-9459-11e6-885d-5f7aeb908be3.png)

as the pics show some lines which case does not trigger in  developer's code is not counted as the denominator . It seems not reasonable.

**Comment by JunelanMe** [comment]
Thanks .
This issue cause bcz of 
https://github.com/sebastianbergmann/php-code-coverage/issues/411 
https://github.com/sebastianbergmann/php-code-coverage/issues/488
https://bugs.xdebug.org/view.php?id=1386

It fixed now in xdebug 2.5.1
